File: 219.cpp
auto find_not_index(std::vector<int>& v, int index, int to_find)
{
    for(int i = 0; i < v.size(); i++)
    {
        if(index != i && to_find == v[i])
        {
            return i;
        }
    }
    return -1;
}

class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        std::map<int, int> counts{};
        for(auto x : nums)
        {
            if(counts.find(x) != counts.end())
            {
                counts[x]++;
            }
            else {
                counts[x] = 1;
            }
        }

        // [1, 2]
        // once we have counts, we need to go through all the counts.
        for(int i = 0; i < nums.size(); i++)
        {
            // 0
            if(counts[nums[i]] > 1)
            {
                if(std::abs(i - find_not_index(nums, i, nums[i])) == k)
                {
                    return true;
                }
            }
        }

        return false;
    }
};

File: 219.py
class Solution(object):
    def containsNearbyDuplicate(self, nums, k):
        dictionary = {}
        for i in range(len(nums)):
            dictionary[nums[i]] = dictionary.get(nums[i], []) + [i]
        for key, value in dictionary.items():
            if len(value) >= 2:
                l,r=0,1
                while r < len(value):
                    if abs(value[r]-value[l]) <= k:
                        return True
                    l,r=l+1,r+1
        return False

        
